corehq/apps/users/forms.py

```diff
@@ -1222,3 +1224,71 @@ def clean_search_string(self):
         if "*" in search_string or "?" in search_string:
             raise forms.ValidationError(_("* and ? are not allowed"))
         return search_string
+
+
+class MobileWorkerAccountConfirmationForm(NoAutocompleteMixin, forms.Form):
+    """
+    For Mobile Workers to confirm their accounts.
+    """
+    # note: this form is heavily inspired by WebUserInvitationForm
+    email = forms.EmailField(label=_('Email Address'),
+                             max_length=User._meta.get_field('email').max_length,
+                             widget=forms.TextInput(attrs={'class': 'form-control'}))
+    full_name = forms.CharField(label=_('Full Name'),
+                                max_length=User._meta.get_field('first_name').max_length +
+                                           User._meta.get_field('last_name').max_length + 1,
+                                widget=forms.TextInput(attrs={'class': 'form-control'}))
+    password = forms.CharField(label=_('Create Password'),
+                               widget=forms.PasswordInput(render_value=False,
+                                                          attrs={
+                                                            'data-bind': "value: password, valueUpdate: 'input'",
+                                                            'class': 'form-control',
+                                                          }),
+                               help_text=mark_safe("""
+                               <span data-bind="text: passwordHelp, css: color">
+                               """))
+    if settings.ENABLE_DRACONIAN_SECURITY_FEATURES:
+        captcha = CaptchaField(_("Type the letters in the box"))
+
+    # required must be set to False to have the clean_*() routine called
+    # todo: should this and other duplicate logic be handled a mixin?
```

Comment:
there's a fair amount of duplication in some of this stuff though it's mostly small. unclear whether it's worth adding - e.g. `EulaConfirmFormMixin` and other such things to encapsulate field-specific logic that is duplicated (also a little unclear to me exactly what the best way to do that is)

Reply:
This is enough duplication that I think it's worth DRYing up. It looks like `WebUserInvitationForm` could be a subclass of `MobileWorkerAccountConfirmationForm`, and all it would add would be `create_domain` (and corresponding logic in `__init__`) and some extra logic in `clean_email`.

Reply:
ok, think this is better https://github.com/dimagi/commcare-hq/pull/26917

Reply:
Haha, good find. Maybe `DomainRegistrationForm` used to have more subclasses.